app/Http/Middleware/TrackRequestMiddleware.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Illuminate\Http\Request;
+use Closure;
+use \OpenTelemetry\API\Trace\SpanInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Illuminate\Log\LogManager;
+use Keepsuit\LaravelOpenTelemetry\Facades\Tracer;
+
+
+class TrackRequestMiddleware
+{
+    /**
+     * @var LogManager
+     */
+    protected LogManager $logger;
+
+    /**
+     * @var float
+     */
+    protected float $startTime = 0;
+
+    /**
+     * @var SpanInterface
+     */
+    protected SpanInterface $span;
+
+    /**
+     * Constructor del middleware.
+     * Laravel inyectará el LogManager aquí.
+     *
+     * @param LogManager $logger
+     */
+    public function __construct(LogManager $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        try {
+            $this->startTime = microtime(true);
+            $this->span = Tracer::newSpan('sample trace')->start();
```

Comment:
@rreynametova what does this 'sample trace' mean ? does need to be hardcded ?
or it could be configurable ?